### PR TITLE
cpan/add_full_name_month_to_date_utility

### PR DESCRIPTION
**Pull request by farzaneh-fs**

### CODE REVIEW

File: t/static_methods.t [hunk 1]
```diff
@@ -31,6 +31,14 @@ is(Date::Utility::month_number_to_abbrev(-1),    undef, '-1 is not a month');
...
 is(Date::Utility::month_number_to_abbrev(13),    undef, '13 is not a month');
 
+# month_number_to_fullname
+is(Date::Utility::month_number_to_fullname(01),    'January', '01 is January');
+is(Date::Utility::month_number_to_fullname(7),     'July', '7 is July');
+is(Date::Utility::month_number_to_fullname(00004), 'April', '00004 is April');
+is(Date::Utility::month_number_to_fullname(-1),    undef, '-1 is not a month');
+is(Date::Utility::month_number_to_fullname(0),     undef, '0 is not a month');
+is(Date::Utility::month_number_to_fullname(13),    undef, '13 is not a month');
+
```
Comment (billmarriott-binary) [inline t/static_methods.t:41]: Please add a test or two for `date_ddmonthyyyy`.

File: lib/Date/Utility.pm [hunk 2]
```diff
@@ -325,6 +327,19 @@ sub _build_date_ddmmmyyyy {
     return join('-', ($self->day_of_month, $self->month_as_string, $self->year));
 }
 
+=head2 date_ddmonthyyyy
```
Comment (billmarriott-binary) [inline lib/Date/Utility.pm:330]: Please add tests for this.

File: lib/Date/Utility.pm [hunk 3]
```diff
@@ -1013,6 +1039,37 @@ sub month_abbrev_to_number {
...
+    11 => 'November',
+    12 => 'December',
+);
+
+sub month_number_to_fullname {
+
+    # Deal with leading zeroes.
+    my $which = int shift;
+
+    return $number_fullname_map{$which};
```
Comment (turnstep) [inline lib/Date/Utility.pm]:
```suggestion
     return $number_fullname_map{int shift};
```

File: t/attributes.t [hunk 4]
```diff
@@ -59,6 +61,7 @@ my %results = (
         date_ddmmyy                 => '02-01-99',
         date_ddmmyyyy               => '02-01-1999',
         date_ddmmmyyyy              => '2-Jan-1999',
+        date_ddmonthyyyy            => '2-January-1999',
```
Comment (kirill-deriv) [inline t/attributes.t]:
```suggestion
        date_ddmonthyyyy            => '2 January 1999',
```

File: t/attributes.t [hunk 5]
```diff
@@ -18,6 +18,7 @@ my %results = (
         date_ddmmyy                 => '06-07-10',
         date_ddmmyyyy               => '06-07-2010',
         date_ddmmmyyyy              => '6-Jul-2010',
+        date_ddmonthyyyy            => '6-July-2010',
```
Comment (kirill-deriv) [inline t/attributes.t]:
```suggestion
        date_ddmonthyyyy            => '6 July 2010',
```

File: t/attributes.t [hunk 6]
```diff
@@ -100,6 +104,7 @@ my %results = (
         date_ddmmyy                 => '17-07-11',
         date_ddmmyyyy               => '17-07-2011',
         date_ddmmmyyyy              => '17-Jul-2011',
+        date_ddmonthyyyy            => '17-July-2011',
```
Comment (kirill-deriv) [inline t/attributes.t]:
```suggestion
        date_ddmonthyyyy            => '17 July 2011',
```

**Pull request merged**